.github/workflows/contracts.yml

```diff
@@ -4,71 +4,90 @@ on:
-        default: 'nightly'
+        description: "Rust toolchain to use"
+        default: "stable"
+        required: false
+        type: string
+      rust-target:
+        description: "Rust target to use"
+        default: "wasm32-unknown-unknown"
         required: false
         type: string
       sc-meta-hash-git:
-        description: 'multiversx-sc-meta git commit hash'
-        default: ''
+        description: "multiversx-sc-meta git commit hash"
+        default: ""
         required: false
         type: string
       sc-meta-version:
-        description: 'multiversx-sc-meta version'
-        default: ''
+        description: "multiversx-sc-meta version"
+        default: ""
         required: false
         type: string
       mx-scenario-go-version:
-        description: 'sc-scenario-go version'
-        default: ''
+        description: "sc-scenario-go version"
+        default: ""
         required: false
         type: string
       path-to-sc-meta:
-        description: 'multiversx-sc-meta from local'
-        default: ''
+        description: "multiversx-sc-meta from local"
+        default: ""
         required: false
         type: string
       clippy-args:
-        description: 'cargo clippy arguments'
-        default: '--all-targets --all-features'
+        description: "cargo clippy arguments"
+        default: "--all-targets --all-features"
         required: false
         type: string
       enable-contracts-size-report:
-        description: 'Enable contracts size report'
+        description: "Enable contracts size report"
         default: true
         required: false
         type: boolean
       enable-interactor-tests:
-        description: 'Enable interactor tests'
+        description: "Enable interactor tests"
         default: false
         required: false
         type: boolean
       coverage-args:
-        description: 'sc-meta test-coverage arguments'
-        default: '--output ./coverage.md'
+        description: "sc-meta test-coverage arguments"
+        default: "--output ./coverage.md"
         required: false
         type: string
     secrets:
       token:
-        description: 'Github token'
+        description: "Github token"
         required: true
 
 jobs:
-  wasm_test:
-    name: Wasm tests
+  setup:
+    name: Setup Environment
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-
       - name: Install rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
           toolchain: ${{ inputs.rust-toolchain }}
-          target: wasm32-unknown-unknown
+          target: ${{ inputs.rust-target }}
+
+      - name: Cache cargo registry and build
+        id: cache-binaries
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin
+            /usr/bin/wasm-opt
+          key: ${{ runner.os }}-binaries-${{ hashFiles('**/Cargo.lock') }}-${{ inputs.sc-meta-version }}-${{ inputs.mx-scenario-go-version }}
```

Comment:
Interesting trick 👍 

I leave this link for reference, for posterity:

https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#hashfiles

Good for repositories whose dependencies does not change often (I see you use the lock files in the cache key), but which trigger a lot of CI workflows.